openstack/baremetalintrospection/v1/introspection/results.go

```diff
@@ -240,6 +247,28 @@ type SystemVendorType struct {
 	SerialNumber string `json:"serial_number"`
 }
 
+// UnmarshalJson interprets an LLDP TLV [key, value] pair as an LldpTlv structure
+func (tlv *LldpTlvType) UnmarshalJSON(data []byte) error {
+	var list []interface{}
+	if err := json.Unmarshal(data, &list); err != nil {
+		return err
+	}
+
+	fieldtype, ok := list[0].(float64)
```

Comment:
It might be a good idea to do a length check first. Maybe something like:

```go
if len(list) != 2 {
    // either return nil and let the field be empty
    // or return an error stating that the JSON structure was incorrect.
    // I'd opt for the former, but up to you.
}
```